In [1]:
import pandas as pd

In [3]:
titanic_df = pd.read_csv("train.csv")
titanic_df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [4]:
titanic_df.shape

(891, 12)

Conditional Based Filtering

In [5]:
mask = titanic_df.Fare > 100
fare_100 = titanic_df[mask]
fare_100.shape

(53, 12)

Out of 891 passangers , only 53 passangers paid more than 100 $ in fare.

In [6]:
female_fare_100 = fare_100[ fare_100.Sex == "female" ]
female_fare_100.shape

(34, 12)

In [7]:
print(f"Out of 53, {round((34/53) * 100, 2)} % are female passengers")
## Why nearly 65% of highest paying customer are female in 1912 ?

Out of 53, 64.15 % are female passengers


In [8]:
fare_100_male = titanic_df[
    (titanic_df.Fare > 100) & (titanic_df.Sex == "male")
]
fare_100_male.shape

(19, 12)

<< Write Description or result >>

In [9]:
# What is Survival Rate of Higher-Class passengers ?

hight_class_survival = titanic_df[
    (titanic_df.Pclass == 1) & (titanic_df.Survived == 1)
]
high_class = titanic_df[(titanic_df.Pclass == 1)]
high_class.shape

(216, 12)

In [10]:
hight_class_survival.shape

(136, 12)